deker_shell/main.py

```diff
@@ -84,16 +86,87 @@ def get_global_coll_variable() -> Collection:
                 pass
 
 
-def start() -> None:
-    """Application entrypoint."""
-    if len(sys.argv) < 2:
-        print("No params passed")
-    elif sys.argv[1].endswith(".py"):
-        runpy.run_path(path_name=sys.argv[1])
+@click.command(context_settings=dict(ignore_unknown_options=True, allow_extra_args=True))
+@click.argument("uri", required=True, type=str)
+@click.option("--workers", type=int, help="Number of threads for Deker.")
+@click.option(
+    "--write-lock-timeout",
+    type=int,
+    help="An amount of seconds " "during which a parallel writing process " "waits for release of the locked file.",
+)
+@click.option(
+    "--write-lock-check-interval",
+    type=int,
+    help="An amount of time (in seconds) "
+    "during which a parallel writing process "
+    "sleeps between checks for locks.",
+)
+@click.option("--loglevel", type=str, help="Level of Deker loggers.")
+@click.option(
+    "--memory-limit",
+    type=str,
+    help="Limit of memory allocation per one array/subset in bytes "
+    "or in human representation of kilobytes, megabytes or gigabytes,"
+    " e.g. '100K', '512M', '4G'."
+    " Human representations will be converted into bytes. "
+    "If result is <= 0, total RAM + total swap is used.",
+)
+@click.pass_context
+def start(
+    ctx: Context,
+    uri: str,
+    workers: Optional[int] = None,
+    write_lock_timeout: Optional[int] = None,
+    write_lock_check_interval: Optional[int] = None,
+    loglevel: Optional[str] = None,
+    memory_limit: Optional[Union[int, str]] = None,
+) -> None:
+    """Application entrypoint.
+
+    :param ctx: click Context with extra parameters (kwargs)
+    :param uri: uri to Deker storage
+    :param workers: number of threads for Deker
+    :param write_lock_timeout: An amount of seconds during which a parallel writing process waits for release
+      of the locked file
+    :param write_lock_check_interval: An amount of time (in seconds) during which a parallel writing process sleeps
+      between checks for locks
+    :param loglevel: Level of Deker loggers
+    :param memory_limit: Limit of memory allocation per one array/subset in bytes or in human representation of
+      kilobytes, megabytes or gigabytes, e.g. ``"100K"``, ``"512M"``, ``"4G"``. Human representations will be
+      converted into bytes. If result is ``<= 0`` - total RAM + total swap is used
+
+      .. note:: This parameter is used for early runtime break in case of potential memory overflow
+    """
+    if uri.endswith(".py"):
+        runpy.run_path(path_name=uri)
```

Comment:
Let's add here conversion of `uri` into Path and check `path.exists()` If False - raise FileNotExistsError